pkg/bootstrap/token.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package bootstrap
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	tokenapi "k8s.io/cluster-bootstrap/token/api"
+	tokenutil "k8s.io/cluster-bootstrap/token/util"
+)
+
+// GeneratesTokenSecret returns a Secret conform to kubeadms bootstrap tokens
+func GenerateTokenSecret(token string, expiration time.Time) (*v1.Secret, error) {
```

Comment:
Sorry for asking but, was this function copied from somewhere? If so, I'd put a reference, mostly as a way to document the logic behind this function (although it's kinda explicit).